### DISCUSSION TITLE
ionic Could not find installation of TypeScript.

### DISCUSSION THREAD

**Discussion by stinggga** [codemagic-ci-cd/codemagic-docs#2549] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Ionic/Cordova

### Steps to reproduce

workflows:
  build:
    name: Build and Export IPA
    scripts:
      - name: Install Capacitor CLI
        script: npm install -g @capacitor/cli

      - name: Sync Capacitor
        script: npx cap sync ios

      - name: Build iOS
        script: |
          cd ios
          xcodebuild -workspace App.xcworkspace -scheme App -configuration Release clean archive -archivePath ${{BUILD_DIR}}/App.xcarchive

      - name: Export IPA
        script: |
          xcodebuild -exportArchive -archivePath ${{BUILD_DIR}}/App.xcarchive -exportOptionsPlist ios/ExportOptions.plist -exportPath ${{BUILD_DIR}}

### Expected results

[error] Could not find installation of TypeScript.
        To use capacitor.config.ts files, you must install TypeScript in your project, e.g. w/ npm install -D typescript

Build failed :|
Step 3 script `Sync Capacitor` exited with status code 1

### Actual results

build shoud be finish

**Comment by dtrdic** [comment]
@stinggga please share a build id so i can check the logs this error description doesn't tell me much.
Please also check an[ example ](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/ionic/ionic-capacitor-demo-project/codemagic.yaml) workflow configuration for Ionic apps as a reference. You can also follow [this guide](https://docs.codemagic.io/yaml-quick-start/building-an-ionic-app/), which will help you set up your yaml workflow properly

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
